src/DirectGateway.php

```diff
@@ -43,6 +44,16 @@ public function setSimulatorMode($value)
         return $this->setParameter('simulatorMode', $value);
     }
 
+    public function getReferrerId()
+    {
+        return $this->getParameter('referrerId');
+    }
+
+    public function setReferrerId($value)
+    {
+        return $this->setParameter('referrerId', $value);
+    }
+
```

Comment:
I tend to shy away from ambiguity, when possible. How would it behave if you were to set it on gateway and request? Does one take precedence? Is it clear that they're the same thing and will over write each other? 

If there's a compelling reason to keep it in both places, that's fine. I just want to make sure that the behavior is well defined, explained and tested for.